packages/component-library/components/Button/components/GenericButton.tsx

```diff
@@ -43,14 +43,39 @@ type Props = ButtonProps & {
   iconButton?: boolean
 }
 
-const GenericButton: React.FunctionComponent<Props> = props => (
-  <span
-    className={classNames(styles.container, {
-      [styles.fullWidth]: props.fullWidth,
-    })}
-  >
-    {props.href ? renderLink(props) : renderButton(props)}
-  </span>
+export type ButtonRef = { focus: () => void }
+
+// We're treating custom props as anything that is kebab cased.
+// This is so we can support properties like aria-* or data-*
+const getCustomProps = (props: object) => {
+  const keys = Object.keys(props).filter(k => k.indexOf("-") !== -1)
+  return keys.reduce((acc, val) => {
+    acc[val] = props[val]
+    return acc
+  }, {})
+}
```

Comment:
I have no idea of the current Kaizen structure, but is it possible to abstract this out?

Reply:
Yeah, I asked the same question 😛 

> It'd be nice to eventually have a package for miscellaneous utils like this. For now, I think this is fine